shared/src/state-migrations/38-deterministic-alarm-groups.ts

```diff
@@ -0,0 +1,304 @@
+    };
+
+    return { vehicle, materials, personnel };
+}
+
+export const deterministicAlarmGroups38: Migration = {
+    action: (intermediaryState, action) => {
+        switch ((action as { type: string }).type) {
+            case '[Vehicle] Add vehicle': {
+                const typedAction = action as VehicleParameters & {
+                    vehicleParameters: VehicleParameters;
+                };
+
+                typedAction.vehicleParameters = {
+                    vehicle: typedAction.vehicle,
+                    materials: typedAction.materials,
+                    personnel: typedAction.personnel,
+                };
+
+                delete typedAction.vehicle;
+                delete typedAction.materials;
+                delete typedAction.personnel;
+
+                break;
+            }
+            case '[Emergency Operation Center] Send Alarm Group': {
+                const typedAction = action as {
+                    alarmGroupId: UUID;
+                    sortedVehicleParameters: VehicleParameters[];
+                    firstVehiclesCount: number;
+                    firstVehiclesTargetTransferPointId: UUID | undefined;
+                };
+                const typedState = intermediaryState as {
+                    materialTemplates: {
+                        [Key in MaterialType]: MaterialTemplate;
+                    };
+                    personnelTemplates: {
+                        [Key in PersonnelType]: PersonnelTemplate;
+                    };
+                    vehicleTemplates: VehicleTemplate[];
+                };
+
+                const alarmGroup = getElement(
+                    intermediaryState as any,
+                    'alarmGroup',
+                    typedAction.alarmGroupId
+                );
+
+                const alarmGroupVehicles = Object.values(
+                    alarmGroup.alarmGroupVehicles as {
+                        [key: UUID]: {
+                            id: UUID;
+                            vehicleTemplateId: UUID;
+                            name: string;
+                        };
+                    }
+                );
+
+                const vehicleTemplatesById = Object.fromEntries(
+                    typedState.vehicleTemplates.map((template) => [
+                        template.id,
+                        template,
+                    ])
+                );
+
+                // We're trying to restore the original vehicle IDs that were generated with `nextUUID`.
+                // Therefore, we must create them in the same order as the original code, i. e., only
+                // sort by time if we have a different destination for the first vehicles
+                if (
+                    typedAction.firstVehiclesCount > 0 &&
+                    typedAction.firstVehiclesTargetTransferPointId
+                ) {
+                    alarmGroupVehicles.sort(
+                        (a: any, b: any) => a.time - b.time
```

Comment:
```suggestion
                        (a: { time: number }, b: { time: number }) => a.time - b.time
```

To give at least some form of guidance here.

Reply:
I added this to the type definition of `alarmGroupVehicles` instead` (see 33159da92ecb94834c48558faeb0508993c25da8)